### PR TITLE
Add `test-gpu` task

### PR DESCRIPTION
**Pull request by Hoxbro**
This should make it easier to test CuDF (locally). I'm using it as a marker as I only want to be able to run the GPU tests if this task is called. 

image

I have used the versions from https://github.com/rapidsai/cudf?tab=readme-ov-file#conda

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -116,6 +116,9 @@ filterwarnings = [
     "ignore:The current Dask DataFrame implementation is deprecated:DeprecationWarning",  # https://github.com/dask/dask/issues/10917
     # 2024-04
     "ignore:No data was collected:coverage.exceptions.CoverageWarning",  # https://github.com/pytest-dev/pytest-cov/issues/627
+    # 2024-05
+    "ignore::numba.core.errors.NumbaPerformanceWarning",  # Not important for tests
+
```
Comment (Hoxbro) [inline pyproject.toml:121]:
```suggestion
```

**Comment by Hoxbro** [comment]
Also added support for `_select_mask_neighbor` as the test was easy to run. 

image

File: holoviews/element/selection.py [hunk 2]
```diff
@@ -83,12 +83,13 @@ def spatial_select_gridded(xvals, yvals, geometry):
...
     if 'cudf' in sys.modules:
         import cudf
+        import cupy as cp
         if isinstance(xvals, cudf.Series):
             xvals = xvals.values.astype('float')
             yvals = yvals.values.astype('float')
             try:
                 import cuspatial
-                result = cuspatial.point_in_polygon(
+                result = cuspatial.point_in_polygon(  # TODO: now only takes two arguments
```
Comment (Hoxbro) [inline holoviews/element/selection.py:92]: @AjayThorve, can you help me get `cuspatial` up and running if you have time?

I get this error `TypeError: point_in_polygon() takes 2 positional arguments but 6 were given` when running the test. I assume this has worked at some point.

Comment (AjayThorve) [reply holoviews/element/selection.py:92]: Hey @Hoxbro. yes there have been some changes in the point_in_polygon api, it only takes two parameters now, both cuspatial GeoSeries (points, polygons). Here is something that might work:

```python
import cudf
import cupy as cp
from shapely.geometry import Polygon
import geopandas

if isinstance(xvals, cudf.Series):
    xvals = xvals.values.astype('float')
    yvals = yvals.values.astype('float')
    try:
        import cuspatial
        df = cudf.DataFrame({'x':xvals, 'y':yvals})
        points = cuspatial.GeoSeries.from_points_xy(
           df.interleave_columns().astype('float')
        )
        polygons = cuspatial.GeoSeries(
           geopandas.GeoSeries(Polygon(geometry)), index=["selection"]
        )

        result = cuspatial.point_in_polygon(points.polygon)

        return result.selection
```

ref: 
1. https://docs.rapids.ai/api/cuspatial/nightly/api_docs/geopandas_compatibility/#cuspatial.GeoSeries

2. https://docs.rapids.ai/api/cuspatial/nightly/user_guide/cuspatial_api_examples/#cuspatial.point_in_polygon

Comment (Hoxbro) [reply holoviews/element/selection.py]: Thank you so much! Worked like a charm.

Do you have a version of when the changes were made?

**Pull request merged**